DataManagementSystem/Agent/RequestOperations/ReplicateAndRegister.py

```diff
@@ -390,11 +395,80 @@ def ftsTransfer(self):
     # Just in case some transfers could not be scheduled, try them with RM
     return self.dmTransfer(fromFTS=True)
 
+  def _checkExistingFTS3Operations(self):
+    """
+       TL;DR: check if there are ongoing FTS3Operation for the current RMS Operation
+
+       Under some conditions, we can be trying to schedule files while
+       there is still an FTS transfer going on. This typically happens
+       when the REA hangs. To prevent further race condition, we check
+       if there are FTS3Operations in a non Final state matching the
+       current operation ID. If so, we put the corresponding files in
+       scheduled mode. We will then wait till the FTS3 Operation performs
+       the callback
+
+       :returns: S_OK with True if we can go on, False if we should stop the processing
+    """
+
+    res = FTS3Client().getOperationsFromRMSOpID(self.operation.OperationID)
+
+    if not res['OK']:
+      self.log.debug(
+          "Could not get FTS3Operations matching OperationID",
+          self.operation.OperationID)
+      return res
+
+    existingFTSOperations = res['Value']
+    # It is ok to have FTS Operations in a final state, so we
+    # care only about the others
+    nonFinishedFTSOperations = [
+        ops for ops in existingFTSOperations if ops.status not in FTS3TransferOperation.FINAL_STATES]
+
+    if not nonFinishedFTSOperations:
+      self.log.debug("No ongoing FTSOperations, all good")
+      return S_OK(True)
+
+    self.log.warn("Some FTS3Operations already exist for the RMS Operation:",
+                  [op.operationID for op in nonFinishedFTSOperations])
+
+    # This would really be a screwed up situation !
+    if len(nonFinishedFTSOperations) > 1:
+      self.log.warn("That's a serious problem !!")
```

Comment:
then why only a warning?

Reply:
because the error is not really here, I don't even know how it could happen. I could remove this altogether